Optimize intro sections in schema guide (#270)
citation-file-format/citation-file-format: schema-guide.md

### Typical example

> For most software however, it is relatively easy to expand the minimal case with some more important information like the version, the date when it was last published, some keywords, etc.:

Reviewer: a bit ambiguous: do you mean "more important" or "more, important" either way I don;t think this addition is needed

Author: Okay, I see, was trying to stress, the importance of having this metadata. Will give this another shot, and if this fails, remove.

Author: Done in https://github.com/citation-file-format/citation-file-format/pull/270/commits/3c3ec0c9f4ac515df99a24e5ae3154103dcbbc88.

version: 0.11.2
```

### Referencing other work

When your software or data builds on what others have already done, it is good practice to add a `references` section to your

> `CITATION.cff` file. This way, you give credit to those on whose work you build your own.

Reviewer: yea I struggled with this one as well in the designs of the revamped cffinit website.

maybe something like
```suggestion
`CITATION.cff` file. This way, you give credit to other works on which your own work is built.
```

Author: Hm, you don't credit works though, but (ideally) people. Also I've realized that both examples suggest that your own work is *completely* built on other work, which seems to neglect own creativity in building a work. How about this:

```suggestion
`CITATION.cff` file. This way, you give credit to the authors of works that your own work builds on.
```

This way you credit the authors, and also "your own work builds on" suggests that you add a lego piece on top of an existing tower?

Reviewer: works for me, I'll commit the suggestion
